winrm/client_test.go

```
@@ -25,3 +112,15 @@ func (s *WinRMSuite) TestClientCreateShell(c *C) {
 	shell, _ := client.CreateShell()
 	c.Assert(shell.ShellId, Equals, "67A74734-DD32-4F10-89DE-49A060483810")
 }
+
+func (s *WinRMSuite) TestClientCreateShellCertAuth(c *C) {
+	client, err := NewClient(&Endpoint{Host: "localhost", Port: 5986, HTTPS: true, Cert: []byte(cert), Key: []byte(key)}, "", "")
+	c.Assert(err, IsNil)
+	client.http = func(client *Client, message *soap.SoapMessage) (string, error) {
+		c.Assert(message.String(), Contains, "http://schemas.xmlsoap.org/ws/2004/09/transfer/Create")
+		return createShellResponse, nil
+	}
+
+	shell, _ := client.CreateShell()
```

Don't swallow errors if it's possible